src/rln.ts

```diff
@@ -126,30 +155,60 @@ export function proofToBytes(p: IRateLimitProof): Uint8Array {
   );
 }
 
+export function poseidonHash(...input: Array<Uint8Array>): Uint8Array {
+  const inputLen = writeUIntLE(new Uint8Array(8), input.length, 0, 8);
+  const lenPrefixedData = concatenate(inputLen, ...input);
+  return zerokitRLN.poseidonHash(lenPrefixedData);
+}
+
+export function sha256(input: Uint8Array): Uint8Array {
+  const inputLen = writeUIntLE(new Uint8Array(8), input.length, 0, 8);
+  const lenPrefixedData = concatenate(inputLen, input);
+  return zerokitRLN.hash(lenPrefixedData);
+}
+
 export class RLNInstance {
   constructor(
     private zkRLN: number,
     private witnessCalculator: WitnessCalculator
   ) {}
 
-  generateMembershipKey(): MembershipKey {
-    const memKeys = zerokitRLN.generateMembershipKey(this.zkRLN);
-    return MembershipKey.fromBytes(memKeys);
+  generateIdentityCredentials(): IdentityCredential {
+    const memKeys = zerokitRLN.generateExtendedMembershipKey(this.zkRLN); // TODO: rename this function in zerokit rln-wasm
+    return IdentityCredential.fromBytes(memKeys);
   }
 
-  generateSeededMembershipKey(seed: string): MembershipKey {
+  generateSeededIdentityCredential(seed: string): IdentityCredential {
     const seedBytes = stringEncoder.encode(seed);
-    const memKeys = zerokitRLN.generateSeededMembershipKey(
+    // TODO: rename this function in zerokit rln-wasm
+    const memKeys = zerokitRLN.generateSeededExtendedMembershipKey(
       this.zkRLN,
       seedBytes
     );
-    return MembershipKey.fromBytes(memKeys);
+    return IdentityCredential.fromBytes(memKeys);
```

Comment:
is there a difference between Identity Credentials and Credentials?

Reply:
Originally `MembershipKey` only had `IDKey` and `IDCommitment` but in https://github.com/waku-org/nwaku/pull/1466 this structure was renamed to IdentityCredential and added 2 new attributes to it (IDTrapdoor and IDNullifier) and renamed `IDKey` to `IDSecretHash`. For the sake of making it easier to compare js-rln with nwaku rln implementation I decided to rename it as well.

Since this is a breaking change I ended up creating a new minor version instead of just bumping up the revision